# Hi, I'm [Ettore](https://github.com/ettorerocchi)!

[![GitHub followers](https://img.shields.io/github/followers/ettorerocchi?label=Followers&style=social)](https://github.com/ettorerocchi) 
[![LinkedIn](https://img.shields.io/badge/LinkedIn-0A66C2?style=flat&logo=linkedin&logoColor=white)](https://www.linkedin.com/in/ettore-rocchi/) 
[![Google Scholar](https://img.shields.io/badge/Google_Scholar-100000?style=flat&logo=Google-Scholar&logoColor=white&labelColor=blue)](https://scholar.google.com/citations?user=MKHoGnQAAAAJ) 
[![Scopus](https://img.shields.io/badge/Scopus-Elsevier-orange)](https://www.scopus.com/authid/detail.uri?authorId=57220152522)

<div align="left">
  <img src="https://media1.tenor.com/m/p4bVBy2jTB8AAAAd/lost-numbers.gif" width="400" alt="Lost Numbers GIF">
</div>


## üßë‚Äçüíª About Me

I am a PhD student at the Department of Medical and Surgical Sciences, University of Bologna (Italy). I am also part of the Multi-Omics and Health-Care Data Analytics Unit at Sant'Orsola hospital (Bologna, Italy). With a background in Physics, I now work at the intersection of **machine learning** and **biomedicine**, focusing on:

- Genomics & metagenomics
- Mass spectrometry
- High-dimensional data analysis and integration

My aim is to develop **interpretable**, **reproducible**, and **scalable** pipelines in the field of _omics_ data, providing data-driven solutions for biological and clinical problem solving.


## üî¨ Research Interests

- **Machine Learning & Deep Learning in Biomedicine**  
  Feature engineering, multi-omics integration, and development of interpretable, generalizable models for clinical and biological prediction tasks.

- **Algorithms & Models for Biomedical Data**  
  Development of computational algorithms, statistical models, and network-based approaches to simulate, analyze, and interpret biological systems.

- **Mass Spectrometry**  
  Machine learning methods for predicting antimicrobial resistance from MALDI-TOF mass spectra.

- **Metagenomics**  
  Advanced biostatistical analysis and network-based modeling of complex microbial communities for microbiome profiling, community simulation, and pathogen detection from environmental and clinical samples.

- **Genomics**  
  Computational and machine learning approaches for analyzing structural variants and APOBEC-style mutations; computational approaches to CRISPR genome editing.


## üîß Featured Projects  

<table>
  <tr>
    <td style="padding:10px">
      <a href="https://github.com/EttoreRocchi/combatlearn">
        <img src="https://github-readme-stats.vercel.app/api/pin/?username=EttoreRocchi&repo=combatlearn&title_color=007BFF&icon_color=007BFF&text_color=cccccc&bg_color=1a1a1a" alt="combatlearn">
      </a>
    </td>
    <td style="padding:10px; color:#cccccc;">
      <b><a href="https://github.com/EttoreRocchi/combatlearn" style="color:#007BFF;">combatlearn</a></b>  
      <br>Scikit-learn compatible implementation of the <code>ComBat</code> algorithm for batch effect correction in high-dimensional data.  
      <br><em>Useful for harmonizing high-throughput data across studies.</em>
    </td>
  </tr>
  <tr>
    <td style="padding:10px">
      <a href="https://github.com/EttoreRocchi/MaldiAMRKit">
        <img src="https://github-readme-stats.vercel.app/api/pin/?username=EttoreRocchi&repo=MaldiAMRKit&title_color=007BFF&icon_color=007BFF&text_color=cccccc&bg_color=1a1a1a" alt="MaldiAMRKit">
      </a>
    </td>
    <td style="padding:10px; color:#cccccc;">
      <b><a href="https://github.com/EttoreRocchi/MaldiAMRKit" style="color:#007BFF;">MaldiAMRKit</a></b>  
      <br>Toolkit to read and preprocess MALDI-TOF mass spectra for AMR analyses.
    </td>
  </tr>
  <tr>
    <td style="padding:10px">
      <a href="https://github.com/Physics4MedicineLab/CATS">
        <img src="https://github-readme-stats.vercel.app/api/pin/?username=Physics4MedicineLab&repo=CATS&title_color=007BFF&icon_color=007BFF&text_color=cccccc&bg_color=1a1a1a" alt="CATS">
      </a>
    </td>
    <td style="padding:10px; color:#cccccc;">
      <b><a href="https://github.com/Physics4MedicineLab/CATS" style="color:#007BFF;">CATS</a></b>  
      <br>A tool to compare Cas9 nucleases by detecting overlapping PAM sites and enabling allele-specific targeting of disease mutations, with ClinVar integration and a user-friendly interface.
      <br><em><a href="https://doi.org/10.3389/fgeed.2025.1571023" style="color:#cccccc;">CATS: a bioinformatic tool for automated Cas9 nucleases activity comparison in clinically relevant contexts</a> by Rocchi et al. (2025).</em>
    </td>
  </tr>
  <tr>
    <td style="padding:10px">
      <a href="https://github.com/EttoreRocchi/ResPredAI">
        <img src="https://github-readme-stats.vercel.app/api/pin/?username=EttoreRocchi&repo=ResPredAI&title_color=007BFF&icon_color=007BFF&text_color=cccccc&bg_color=1a1a1a" alt="ResPredAI">
      </a>
    </td>
    <td style="padding:10px; color:#cccccc;">
      <b><a href="https://github.com/EttoreRocchi/ResPredAI" style="color:#007BFF;">ResPredAI</a></b>  
      <br>Machine learning pipeline for predicting antimicrobial resistance from demographic and clinical data.  
      <br><em><a href="https://doi.org/10.1038/s41746-025-01696-x" style="color:#AAAAFF;">Artificial Intelligence model to predict resistances in Gram-negative bloodstream infections</a> by Bonazzetti et al. (2025).</em>
    </td>
  </tr>
  <tr>
    <td style="padding:10px">
      <a href="https://github.com/Physics4MedicineLab/CAMISIM-BrokenStick">
        <img src="https://github-readme-stats.vercel.app/api/pin/?username=Physics4MedicineLab&repo=CAMISIM-BrokenStick&title_color=007BFF&icon_color=007BFF&text_color=cccccc&bg_color=1a1a1a" alt="CAMISIM-BrokenStick">
      </a>
    </td>
    <td style="padding:10px; color:#cccccc;">
      <b><a href="https://github.com/Physics4MedicineLab/CAMISIM-BrokenStick" style="color:#007BFF;">CAMISIM-BrokenStick</a></b>  
      <br>Extension of <code>CAMISIM</code> to simulate metagenomic sequencing data using a broken stick model.  
      <br><em>Enables simulation of metagenomic data with controllable evolutionary dynamics of microbial communities.</em>
    </td>
  </tr>
  <tr>
  <td style="padding:10px">
    <a href="https://github.com/Physics4MedicineLab/APOBECSeeker">
      <img src="https://github-readme-stats.vercel.app/api/pin/?username=Physics4MedicineLab&repo=APOBECSeeker&title_color=007BFF&icon_color=007BFF&text_color=cccccc&bg_color=1a1a1a" alt="APOBECSeeker">
    </a>
  </td>
  <td style="padding:10px; color:#cccccc;">
    <b><a href="https://github.com/Physics4MedicineLab/APOBECSeeker" style="color:#007BFF;">APOBECSeeker</a></b>  
    <br>Pipeline for the identification of APOBEC-style mutations from multiple sequence alignment.
    <br><em><a href="https://doi.org/10.1038/s41598-025-11855-5" style="color:#cccccc;">Understanding the evolutionary dynamics of Monkeypox virus through less explored pathways</a> by De Pascali et al. (2025).</em>
  </td>
</tr>

</table>

## Organization
I am a member of the [Physics4MedicineLab](https://github.com/Physics4MedicineLab), the Applied Physics group within the Department of Medicine at Alma Mater Studiorum - University of Bologna.

## üì´ Connect with Me  
- [LinkedIn](https://www.linkedin.com/in/ettore-rocchi/)  
- [University profile](https://www.unibo.it/sitoweb/ettore.rocchi3)  
- [Email](mailto:mailto:ettore.rocchi3@unibo.it)
